pkg/bloomgateway/client.go

```diff
@@ -211,8 +220,15 @@ func (c *GatewayClient) FilterChunks(ctx context.Context, _ string, interval blo
 	servers := make([]addrWithGroups, 0, len(blocks))
 	for _, blockWithSeries := range blocks {
 		addr, err := c.pool.Addr(blockWithSeries.block.String())
+
+		// the client should return the full, unfiltered list of chunks instead of an error
 		if err != nil {
-			return nil, errors.Wrapf(err, "server address for block: %s", blockWithSeries.block)
+			level.Error(c.logger).Log("msg", "failed to resolve server address for block", "block", blockWithSeries.block, "err", err)
```

Comment:
I don't think resolving a single address should necessarily fail all addresses, but gracefully degrading the whole request is a fair place to start.

Reply:
There is only one possible ways how this can fail, because there is only one case `Addr()` returns an error, and this is because there are no servers available.
Therefore subsequent calls of `Addr()` are **very likely** to fail as well.